mpas_analysis/ocean/nino34_index.py

```diff
@@ -141,15 +141,19 @@ def compute_nino34_index(regionSST, calendar):  # {{{
     03/23/2017
     """
 
-    if not isinstance(regionSST, xr.core.dataarray.DataArray):
-        raise ValueError('regionSST should be an xarray DataArray')
+    if not isinstance(ds, xr.core.dataset.Dataset):
+        raise ValueError('ds should be an xarray Dataset')
+
+    ds = climatology.add_months_and_days_in_month(ds, calendar)
 
     # Compute monthly average and anomaly of climatology of SST
-    monthlyClimatology = climatology.compute_monthly_climatology(regionSST,
-                                                                 calendar)
-    anomalySST = regionSST.groupby('month') - monthlyClimatology
+    monthlyClimatology = climatology.compute_monthly_climatology(ds)
+
+    print monthlyClimatology
+
+    anomaly = ds.groupby('month') - monthlyClimatology
 
-    return _running_mean(anomalySST.to_pandas())  # }}}
+    return _running_mean(anomaly.avgSurfaceTemperature.to_pandas())  # }}}
```

Comment:
`avgSurfaceTemperature` is explicitly selected as the only variable in `ds` when `ds` is constructed, so I don't think that's a major concern.  I agree that it would be clearer if we simply operated on a data array ('regionSST`) as before. I will expound on why I didn't have any success with that approach below and maybe we can find a solution.